### PR TITLE
Support PyTree API for Wires objects

### PR DESCRIPTION
**Pull request by grwlf**
**Context:**

In order to pass objects to Catalyst (`@qjit`)-compiled functions, objects should be Jax-compatible or support the PyTree serialization API. The issue https://github.com/PennyLaneAI/catalyst/issues/334 highlighted the fact that Wires class does not match these requirements, as well as the use-case where it could actually be useful.

**Description of the Change:**

In this PR we add the PyTree serialization API to the `Wires` class.
  
**Benefits:**

Wires objects could be passed as a positional argument to a Catalyst-compiled function.

**Possible Drawbacks:**
- Not known

**Related GitHub Issues:**
- https://github.com/PennyLaneAI/catalyst/issues/334

[sc-48585]

### CODE REVIEW

**Review by albi3ro — COMMENTED** [review]
changelog entry?

**Comment by grwlf** [comment]
@albi3ro I have added one, thanks.

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -26,4 +29,5 @@
 This release contains contributions from (in alphabetical order):
 
 Abhishek Abhishek,
-Matthew Silverman.
\ No newline at end of file
+Matthew Silverman,
+Sergei Mironov.
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:33]:
```suggestion
Sergei Mironov,
Matthew Silverman,
```

**Review by albi3ro — APPROVED** [review]
I'm good with this change for pennylane, and it shouldn't hurt anything.

File: doc/releases/changelog-dev.md (hunk 1 above)
Comment (grwlf) [reply doc/releases/changelog-dev.md:33]: Not important for me, but does the comment says it should be in alphabetical order?

**Comment by grwlf** [comment]
@albi3ro Could you recommend the second reviewer for this PR?
Also, should I re-target this PR to the release branch? Could you please tell me which one?

File: doc/releases/changelog-dev.md (hunk 1 above)
Comment (grwlf) [reply doc/releases/changelog-dev.md]: Not relevant anymore: release branch has a different changelog file.

**Review by dime10 — APPROVED** [review]
Thanks @grwlf :)

**Pull request merged**